ext/flash/01_http.js

```diff
@@ -353,183 +540,37 @@
-                  typeof respBody === "string" ||
-                  ObjectPrototypeIsPrototypeOf(Uint8ArrayPrototype, respBody)
-                );
-              } else {
-                if (innerResp.body.streamOrStatic.consumed === true) {
-                  throw new TypeError("Body is unusable.");
-                }
-                innerResp.body.streamOrStatic.consumed = true;
-                respBody = innerResp.body.streamOrStatic.body;
-              }
-            } else {
-              respBody = new Uint8Array(0);
-            }
-
-            const ws = resp[_ws];
-            if (isStreamingResponseBody === false) {
-              const length = respBody.byteLength || core.byteLength(respBody);
-              const responseStr = http1Response(
-                method,
-                innerResp.status ?? 200,
-                innerResp.headerList,
-                respBody,
-                length,
-              );
-              writeFixedResponse(
-                serverId,
-                i,
-                responseStr,
-                length,
-                !ws, // Don't close socket if there is a deferred websocket upgrade.
-                respondFast,
-              );
-            }
-
-            (async () => {
-              if (!ws) {
-                if (hasBody && body[_state] !== "closed") {
-                  // TODO(@littledivy): Optimize by draining in a single op.
-                  try {
-                    await req.arrayBuffer();
-                  } catch { /* pass */ }
-                }
-              }
-
-              if (isStreamingResponseBody === true) {
-                const resourceRid = getReadableStreamRid(respBody);
-                if (resourceRid) {
-                  if (respBody.locked) {
-                    throw new TypeError("ReadableStream is locked.");
-                  }
-                  const reader = respBody.getReader(); // Aquire JS lock.
-                  try {
-                    core.opAsync(
-                      "op_flash_write_resource",
-                      http1Response(
-                        method,
-                        innerResp.status ?? 200,
-                        innerResp.headerList,
-                        0, // Content-Length will be set by the op.
-                        null,
-                        true,
-                      ),
-                      serverId,
-                      i,
-                      resourceRid,
-                    ).then(() => {
-                      // Release JS lock.
-                      readableStreamClose(respBody);
-                    });
-                  } catch (error) {
-                    await reader.cancel(error);
-                    throw error;
-                  }
-                } else {
-                  const reader = respBody.getReader();
-                  writeFixedResponse(
+              resp = handler(req);
+              if (resp instanceof Promise || resp.then) {
```

Comment:
In what situation is something not a promise but has a `.then` method?

Reply:
It seems return value like this would trigger this conditional but it shouldn't:

```js
return {
   // ...
   then: 1
}
```